src/main/thrift/parquet.thrift

```diff
@@ -753,6 +753,9 @@ struct ColumnMetaData {
 
   /** Byte offset from beginning of file to Bloom filter data. **/
   14: optional i64 bloom_filter_offset;
+
+  /** Size of Bloom filter data, in bytes. **/
+  15: optional i32 bloom_filter_length;
```

Comment:
Seems that if length exists, offset must exists. However, if offset not exists, length should not exist?

And reader should check that:
1. Does it have offset
2. If has, can it go fast path that using bloom_filter_length to read?

Reply:
On the writer side:
- Old writer only writes offset.
- New writer should write length as well.

On the reader size:
- Old reader only checks offset.
- New reader checks offset then try to use length if exists.

Reply:
for future implementers we should add a comment noting that existing of one element does not indicate existence of the other (even though it is already implied by mapping).

Reply:
Agreed.

I have thought to add a comment to say that `bloom_filter_length` is introduced by v2.10.0 (the next unreleased version) and readers may not expect its existence from legacy files or some implementations.